src/bin/pgcopydb/dump_restore.c

```diff
@@ -110,46 +153,24 @@ copydb_dump_source_schema(CopyDataSpec *specs,
 		return false;
 	}
 
-	if (section == PG_DUMP_SECTION_SCHEMA ||
-		section == PG_DUMP_SECTION_PRE_DATA ||
-		section == PG_DUMP_SECTION_ALL)
-	{
-		if (file_exists(specs->dumpPaths.preFilename))
-		{
-			log_info("Skipping pg_dump --section=pre-data, "
-					 "as \"%s\" already exists",
-					 specs->dumpPaths.preFilename);
-		}
-		else if (!pg_dump_db(&(specs->pgPaths),
-							 &(specs->connStrings),
-							 snapshot,
-							 "pre-data",
-							 &(specs->filters),
-							 &(specs->catalogs.filter),
-							 specs->dumpPaths.preFilename))
-		{
-			/* errors have already been logged */
-			return false;
-		}
-	}
+	char *sections[3] = { NULL };
+	getSectionOptions(section, sections);
 
-	if (section == PG_DUMP_SECTION_SCHEMA ||
-		section == PG_DUMP_SECTION_POST_DATA ||
-		section == PG_DUMP_SECTION_ALL)
+	if (sections[0])
 	{
-		if (file_exists(specs->dumpPaths.postFilename))
+		if (file_exists(specs->dumpPaths.dumpFilename))
 		{
-			log_info("Skipping pg_dump --section=post-data, "
+			log_info("Skipping pg_dump --section=pre-data, "
 					 "as \"%s\" already exists",
-					 specs->dumpPaths.postFilename);
+					 specs->dumpPaths.dumpFilename);
 		}
 		else if (!pg_dump_db(&(specs->pgPaths),
 							 &(specs->connStrings),
 							 snapshot,
-							 "post-data",
+							 sections,
```

Comment:
For pg_dump, it's going to be `--schema-only` always, so I believe this could be a static information in the `pg_dump_db` implementation?